BDArmory/Weapons/Missiles/MissileBase.cs

```diff
@@ -1648,7 +1648,8 @@ public void CheckDetonationState(bool separateWarheads = false)
                                     Vector3 relPos = TargetPosition - vessel.CoM;
                                     Vector3 relVel = TargetVelocity - vessel.Velocity();
                                     bool approaching = Vector3.Dot(relPos, relVel) < 0f;
-                                    float targetRad = targetVessel.Vessel.GetRadius();
+                                    float targetRad = heatTarget.exists && heatTarget.vessel == null ? 1 : targetVessel.Vessel.GetRadius();
```

Comment:
What happens if `heatTarget.vessel` is an actual vessel, e.g., switching between two closely flying planes? Shouldn't `targetVessel` then get set on lines 758 and 1966?

Reply:
Would depend on how we want the AI missile launch logic to work. The primary use iirc of missiles tracking what vessel they're homing in on is for the MissilesAway accounting in MissileFire, so the AI knows how many missiles have been fired on target x, compared to its max missiles per target setting; do we want the AI to magically know that the missile it's fired at target X has gotten decoyed and is now going after target Y, and immediately fire another missile at X to bring the number of missiles in the air back to maxMissilesPerTarget?

Reply:
OK, so `targetVessel` here is the vessel the missile was originally fired against and what the AI thinks it's targeting.
That makes more sense that it wouldn't be updated once the AI has fired it then, unless its capable of communicating with the firing vessel when its lock is updated. I see that radar missiles can do this on line 899
```csharp
targetVessel = t.targetInfo; //reset targetvessel in case of canRelock getting a new target
```
(it's the only reference to `targetVessel.set` in MissileBase that doesn't doesn't set it to `null`) via `vrd`, which I presume is fine?

Reply:
In the radar missile case, line 899 is when the missile is SARH and being actively guided by the parent craft radar painting the target, so updating the targetvessel there makes sense.